Upgrade documentation for testers

GiudGiud opened the pull request:
- add custom cmp explanation to CSVDiff
- fix variable names in exodiff
- add custom cmp for exodiff in exodiff instead of for csvdiff in exodiff refs #19983

Ran into this while fixing the VTB. This could be a tad clearer

GiudGiud committed:
Upgrade documentation for testers:
- add custom cmp explanation to CSVDiff
- fix variable names in exodiff
- add custom cmp for exodiff in exodiff instead of for csvdiff in exodiff
refs #19983

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1246886/) on a7f36d4 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/22811/site)

This comment will be updated on new commits.

cticenhour commented on modules/doc/content/python/testers/Exodiff.md:
```diff
@@ -10,10 +10,10 @@ Complete documentation for the `Exodiff` utility may be found [here](https://san
   be met for a test to pass.
 
-- `rel_tol`: sets a relative tolerance. Defaults to 5.5e-6.
+- `rel_err`: sets a relative tolerance. Defaults to 5.5e-6.
```
```suggestion
- `rel_err`: sets a relative error tolerance. Defaults to 5.5e-6.
```

cticenhour commented on modules/doc/content/python/testers/Exodiff.md:
```diff
@@ -10,10 +10,10 @@ Complete documentation for the `Exodiff` utility may be found [here](https://san
 Test configuration options are added to the `tests` file.
 
-- `abs_tol`: sets an absolute tolerance, defaults to 1e-10. Both absolute and relative tolerances must
+- `abs_zero`: sets an absolute tolerance, defaults to 1e-10. Both absolute and relative tolerances must
   be met for a test to pass.
```
```suggestion
- `abs_zero`: sets an absolute zero cutoff used in comparisons, defaults to 1e-10.
```

cticenhour commented on modules/doc/content/python/testers/JSONDiff.md:
```diff
@@ -12,6 +12,11 @@ Test configuration options are added to the `tests` file.
 
 - `skip_keys`: A list of keys to skip in the `JSON` comparison
 
+- `abs_zero`: Sets an absolute tolerance, defaults to 1e-10. Both absolute and relative tolerances must
+  be met for a test to pass.
+
+- `rel_err`: Sets a relative tolerance, defaults to 5.5e-6.
+
```
```suggestion
- `abs_zero`: Sets an absolute zero cutoff used in comparisons, defaults to 1e-10.

- `rel_err`: Sets a relative error comparison tolerance, defaults to 5.5e-6.

```

cticenhour commented on modules/doc/content/python/testers/Exodiff.md:
```diff
@@ -42,17 +42,13 @@ ${MOOSE_DIR}/framework/contrib/exodiff/exodiff -summary moose_simulation_out.e
-comparison for that field.
+In the example below, the tolerances are adjusted for local comparisons of field variables,
+both nodal and elemental. The relative error criteria is set to `1e-6` and the absolute
+criteria is disabled by setting it to `0`. These modifications apply to variable `u` and `distance`,
```
By "absolute criteria" do you mean the floor parameter? This effectively sets the absolute zero parameter for comparisons on a per group basis. Having it set to 0 is the exodiff default for variables and then 1e-15 for timesteps (per the [Exodiff doco](https://sandialabs.github.io/seacas-docs/exo_util.pdf#page=11)). 

If floor is what you were referring to, this statement should be tweaked.

GiudGiud replied:
```suggestion
criteria is strongly tightened by setting it to `0`. These modifications apply to variable `u` and `distance`,
```

cticenhour reviewed (changes requested):
Thanks once again for the improvements here - a few comments.

GiudGiud committed:
Apply suggestions from code review

Clearer formulations of what each error criteria is 
Fix the explanation for the 'absolute' error criteria being set to 0

Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>

GiudGiud commented:
should be ready for another look

cticenhour reviewed (approved):
Thanks for clarifying your intention with the phrasing in that follow-up commit - looks good now.
